src/main/java/www/purple/mixxy/helpers/GoogleAuthHelper.java

```diff
@@ -0,0 +1,104 @@
+package www.purple.mixxy.helpers;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeRequestUrl;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.Collection;
+
+public final class GoogleAuthHelper {
+
+	/**
+	 * Please provide a value for the CLIENT_ID constant before proceeding, set this up at https://code.google.com/apis/console/
+	 */
+	private static final String CLIENT_ID = "734052615495-2751ophn3s30pvldar5j7tkicgp7ga2s.apps.googleusercontent.com";
+	/**
+	 * Please provide a value for the CLIENT_SECRET constant before proceeding, set this up at https://code.google.com/apis/console/
+	 */
+	private static final String CLIENT_SECRET = "SXQxxtOPo3gMxtuW_bcaHI_O";
+
+	/**
+	 * Callback URI that google will redirect to after successful authentication
+	 */
+	private static final String CALLBACK_URI = "http://localhost:8080/validate";
```

Comment:
There's got to be a better way to do this than by just slapping `localhost` in there.